data/structures.yml

```diff
@@ -892,7 +892,7 @@ CA:
   :account_number_length: 12
   :bank_code_format: "\\d{4}"
   :branch_code_format: "\\d{5}"
-  :account_number_format: "\\d{7,12}"
+  :account_number_format: "\\A\\d{7,12}\\z"
```

Comment:
It feels like the _intention_ of all the other formats in this file is the same as this, so maybe we should be applying this in the code, rather than making this something that has to be remembered in this config file?

Reply:
Makes sense, ill update it to check the full string

Reply:
Made this change - it worked mostly as expected but Belgium had an invalid `account_number_format` afterwards I made the change to accept the format but can you give it a look over. 

Thanks!

Reply:
Agree with @isaacseymour but he beat me to it